README.md

```diff
@@ -194,6 +194,26 @@ https://docs.ansible.com/ansible/latest/user_guide/vault.html for details.
 If these variables are set, `ha_cluster_regenerate_keys` is ignored for this
 certificate - key pair.
 
+#### `ha_cluster_pcsd_certificates`
+
+You can either use existing TLS certificate and private key files by providing
+them with [`ha_cluster_pcsd_public_key_src` and `ha_cluster_pcsd_private_key_src`](#ha_cluster_pcsd_public_key_src-ha_cluster_pcsd_private_key_src),
+or use the role to create certificates by providing `ha_cluster_pcsd_certificates`.
+
+Use this variable to generate certificate and private key for TLS encryption
+using the `fedora.linux_system_roles.certificate`.
+
+The value of `ha_cluster_pcsd_certificates` is set to the variable `certificate_requests`
+in the `certificate` role.
+For more information, see the `certificate_requests` section in the `certificate`
+role documentation.
+
+When you set this variable, you must not set `ha_cluster_pcsd_public_key_src` and
+`ha_cluster_pcsd_private_key_src` variables.
```

Comment:
It would be nice if you could add a check for this into `check-and-prepare-role-variables.yml`.

Reply:
In the current behavior, if `ha_cluster_pcsd_public_key_src` and `ha_cluster_pcsd_private_key_src` variables are set along with `ha_cluster_pcsd_certificates`, the first two variables are just ignored. Do we still want to make the case `fail`? It's a bit tricky. It does not do any harm even if all 3 variables are set, but it may surprise the customer if they expect their existing cert and key will not be copied to the managed node. What do you think, @tomjelinek?

Reply:
> It does not do any harm even if all 3 variables are set, but it may surprise the customer if they expect their existing cert and key will not be copied to the managed node. What do you think, @tomjelinek?

Yes, that's my thinking. The role would not fail, but the outcome may not match the expectations.

The role already has many input variables. It seems quite possible to set `ha_cluster_pcsd_public_key_src` and `ha_cluster_pcsd_private_key_src` and forget to delete `ha_cluster_pcsd_certificates`. In that case, documentation doesn't help. Actively checking and failing the role does.

I'm not sure if such a check is a best practice, though. Until now, `check-and-prepare-role-variables.yml` looks for such variables configurations which would fail the role anyway.